cli/command/stack/deploy.go

```diff
@@ -30,68 +41,165 @@ func newDeployCommand(dockerCli *command.DockerCli) *cobra.Command {
+		fmt.Fprintf(dockerCli.Err(), "Ignoring deprecated options:\n\n%s\n\n",
+			propertyWarnings(deprecatedProperties))
 	}
 
-	networks := getUniqueNetworkNames(bundle.Services)
 	ctx := context.Background()
+	namespace := namespace{name: opts.namespace}
 
-	if err := updateNetworks(ctx, dockerCli, networks, opts.namespace); err != nil {
+	networks := convertNetworks(namespace, config.Networks)
+	if err := createNetworks(ctx, dockerCli, namespace, networks); err != nil {
+		return err
+	}
+	services, err := convertServices(namespace, config)
+	if err != nil {
 		return err
 	}
-	return deployServices(ctx, dockerCli, bundle.Services, opts.namespace, opts.sendRegistryAuth)
+	return deployServices(ctx, dockerCli, services, namespace, opts.sendRegistryAuth)
 }
 
-func getUniqueNetworkNames(services map[string]bundlefile.Service) []string {
-	networkSet := make(map[string]bool)
-	for _, service := range services {
-		for _, network := range service.Networks {
-			networkSet[network] = true
-		}
+func propertyWarnings(properties map[string]string) string {
+	var msgs []string
+	for name, description := range properties {
+		msgs = append(msgs, fmt.Sprintf("%s: %s", name, description))
+	}
+	sort.Strings(msgs)
+	return strings.Join(msgs, "\n\n")
+}
+
+func getConfigDetails(opts deployOptions) (composetypes.ConfigDetails, error) {
+	var details composetypes.ConfigDetails
+	var err error
+
+	details.WorkingDir, err = os.Getwd()
+	if err != nil {
+		return details, err
+	}
+
+	configFile, err := getConfigFile(opts.composefile)
+	if err != nil {
+		return details, err
 	}
+	// TODO: support multiple files
+	details.ConfigFiles = []composetypes.ConfigFile{*configFile}
+	return details, nil
+}
 
-	networks := []string{}
-	for network := range networkSet {
-		networks = append(networks, network)
+func getConfigFile(filename string) (*composetypes.ConfigFile, error) {
+	bytes, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return nil, err
+	}
+	config, err := loader.ParseYAML(bytes)
+	if err != nil {
+		return nil, err
 	}
-	return networks
+	return &composetypes.ConfigFile{
+		Filename: filename,
+		Config:   config,
+	}, nil
 }
 
-func updateNetworks(
+func convertNetworks(
+	namespace namespace,
+	networks map[string]composetypes.NetworkConfig,
+) map[string]types.NetworkCreate {
+	if networks == nil {
+		networks = make(map[string]composetypes.NetworkConfig)
+	}
+
+	// TODO: only add default network if it's used
+	networks["default"] = composetypes.NetworkConfig{}
+
+	result := make(map[string]types.NetworkCreate)
+
+	for internalName, network := range networks {
+		if network.External.Name != "" {
+			continue
+		}
+
+		createOpts := types.NetworkCreate{
+			Labels:  getStackLabels(namespace.name, network.Labels),
+			Driver:  network.Driver,
+			Options: network.DriverOpts,
+		}
+
+		if network.Ipam.Driver != "" {
+			createOpts.IPAM = &networktypes.IPAM{
+				Driver: network.Ipam.Driver,
+			}
+		}
+		// TODO: IPAMConfig.Config
```

Comment:
we only have subnet implemented in `compose-file`. I will implement that, but the rest will have to wait.

I'm not sure if swarm mode supports all of this, I remember hearing some of it is ignored.